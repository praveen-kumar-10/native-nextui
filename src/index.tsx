import React from 'react';
import { Pressable, StyleSheet, Text, type PressableProps } from 'react-native';

interface ButtonProps extends PressableProps {
  children: React.ReactNode;
}

export const Button = ({ children, ...props }: ButtonProps) => {
  return (
    <Pressable style={styles.wrapper} {...props}>
      <Text style={styles.label}>{children}</Text>
    </Pressable>
  );
};

const styles = StyleSheet.create({
  wrapper: {
    paddingHorizontal: 16,
    paddingVertical: 10,
    backgroundColor: '#006fee',
    borderRadius: 8,
  },
  label: {
    color: '#ffffff',
    fontSize: 20,
  },
});
