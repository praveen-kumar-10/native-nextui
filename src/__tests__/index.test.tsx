import React from 'react';
import { render } from '@testing-library/react-native';

import { Button } from '../../src';

describe('Button', () => {
  it('renders correctly', () => {
    const tree = render(<Button>Text Button</Button>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
