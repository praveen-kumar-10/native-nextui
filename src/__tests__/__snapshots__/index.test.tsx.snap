// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#006fee",
      "borderRadius": 8,
      "paddingHorizontal": 16,
      "paddingVertical": 10,
    }
  }
>
  <Text
    style={
      {
        "color": "#ffffff",
        "fontSize": 20,
      }
    }
  >
    Text Button
  </Text>
</View>
`;
